src/containerregistry/docker_keychain.go

```diff
@@ -0,0 +1,82 @@
+package containerregistry
+
+import (
+	"sync"
+
+	"github.com/docker/cli/cli/config"
+	"github.com/docker/cli/cli/config/configfile"
+	dockertypes "github.com/docker/cli/cli/config/types"
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/spf13/afero"
+)
+
+type dockerKeychain struct {
+	mutex            sync.Mutex
+	dockerConfigFile string
+	filesystem       afero.Fs
+}
+
+func NewDockerKeychain(dockerconfigFile string, filesystem afero.Fs) authn.Keychain {
+	return &dockerKeychain{
+		dockerConfigFile: dockerconfigFile,
+		filesystem:       filesystem,
+	}
+}
+
+// Resolve implements Keychain interface by interpreting the docker config file.
+// It is based on the 'defaultKeychain' type from the go-gontainerregistry library
+// https://github.com/google/go-containerregistry/blob/27a6ad6/pkg/authn/keychain.go
+// dockerKeychain implementation can read a docker config file of any name and from any directory.
+func (dk *dockerKeychain) Resolve(target authn.Resource) (authn.Authenticator, error) {
```

Comment:
`dk` is still sometimes used as a short name for the `dynakube` so I would avoid using it here, might cause confusion 😅 

you could used `keychain` or `key` if you want to be short

Reply:
fixed